### ISSUE TITLE
`ignore-paths` Can't use escape sequence in toml config

### ISSUE THREAD

**Issue by Avasam** [PyCQA/pylint#6963]
### Bug description

I'm unable to use an escape sequence to have a literal character in `pyproject.toml`

### Configuration

```ini
[tool.pylint.MASTER]
ignore-paths = [
  # Works
  "^.*.pyi$",
  # Expected to work, does not
  "^.*\\.pyi$",
  # Invalid toml `Failed to load 'pyproject.toml': Unescaped "\" in a string`
  "^.*\.pyi$",
  "^.*\\\.pyi$",
  # Doesn't work
  "^.*\\\\.pyi$",
]
```

### Command used

```shell
`"C:\Program Files\Python39\python.exe" ~\.vscode\extensions\ms-python.python-2022.8.0\pythonFiles\linter.py -m pylint .\typings\pygrabber\dshow_graph\__init__.pyi
cwd: .`
(ran by the vscode python extension)

manually:
`pylint typings/pygrabber/dshow_graph/__init__.pyi`
```

### Pylint output

```shell
From vscode
##########Linting Output - pylint##########

[
// Tons of pylint errors that I wanted to ignore
]


Manual run
************* Module __init__
// Tons of pylint errors that I wanted to ignore

----------------------------------------------------------------------
Your code has been rated at -25.08/10 (previous run: -25.08/10, +0.00)
```

### Expected behavior

```toml
[tool.pylint.MASTER]
ignore-paths = [
  "^.*\\.pyi$",
]
```

Should ignore all .pyi files

### Pylint version

```shell
pylint 2.14.1
astroid 2.11.5
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10.0.19044

**Comment by Avasam** [comment]
I didn,t see that one originally, might be a duplicate of #5398

**Comment by DetachHead** [comment]
you can use a single quoted string to avoid this issue
```toml
[tool.pylint.MASTER]
ignore-paths = [
  '^.*.pyi$',
  '^.*\\.pyi$',
  '^.*\.pyi$',
  '^.*\\\.pyi$',
  '^.*\\\\.pyi$',
]
```
